DependencyInjection/PrivacyCookiePass.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/**
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+namespace EzSystems\DemoBundle\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+
+class PrivacyCookiePass implements CompilerPassInterface
+{
+    public function process( ContainerBuilder $container )
+    {
+        if ( $container->hasDefinition( 'ez_privacy_cookie.twig.extension' ) === false )
+        {
+            return;
+        }
+
+        if ( $container->hasDefinition( 'ezdemo.ez_content_banner_factory' ) === false )
+        {
+            return;
+        }
+
+        $definition = $container->getDefinition( 'ez_privacy_cookie.twig.extension' );
+        $definition->replaceArgument( 1, $container->getDefinition( 'ezdemo.ez_content_banner_factory' ) );
```

Comment:
I wouldn't use a compiler pass for this. There is a service alias that is meant for this exactly:

```yaml
ez_privacy_cookie.banner_factory:
    class: %ez_privacy_cookie.banner_factory.class%
    alias: ez_privacy_cookie.configuration_banner_factory
```

Just redefine this alias in the DemoBundle's `service.yml` file:
```yaml
ez_privacy_cookie.banner_factory:
    class: %ez_privacy_cookie.banner_factory.class%
    alias: ezdemo.ez_content_banner_factory
```